File: bridges/python/src/sdk/tools/schemas/transcription_schema.py
from typing import TypedDict, List, Union


class TranscriptionSegment(TypedDict):
    # Start time in seconds (using from_ since 'from' is a Python keyword)
    from_: float

    # End time in seconds
    to: float

    # Transcribed text for this segment
    text: str

    # Speaker identifier
    speaker: Union[str, None]


class TranscriptionMetadata(TypedDict):
    # Tool that generated the transcription
    tool: str


class TranscriptionOutput(TypedDict):
    # Total audio duration in seconds
    duration: float

    # List of unique speaker identifiers
    speakers: List[str]
    
    # Number of unique speakers
    speaker_count: int

    # Array of transcription segments
    segments: List[TranscriptionSegment]

    # Additional metadata about the transcription
    metadata: TranscriptionMetadata

File: bridges/nodejs/src/sdk/tools/schemas/transcription-schema.ts
export interface TranscriptionSegment {
  // Start time in seconds
  from: number

  // End time in seconds
  to: number

  // Transcribed text for this segment
  text: string

  // Speaker identifier
  speaker: string | null
}

export interface TranscriptionOutput {
  // Total audio duration in seconds
  duration: number

  // List of unique speaker identifiers
  speakers: string[]

  // Number of unique speakers
  speaker_count: number

  // Additional metadata about the transcription
  metadata: {
    // Tool that generated the transcription
    tool: string
  }

  // Array of transcription segments
  segments: TranscriptionSegment[]
}
